src/main/kotlin/app/revanced/patches/twitter/misc/links/ChangeLinkSharingDomainPatch.kt

```diff
@@ -0,0 +1,142 @@
+            }!!.index
+
+            val instruction = this.mutableMethod.getInstruction(stringIndex)
+            val overrideRegister = (instruction as OneRegisterInstruction).registerA
+            val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+
+            var overrideString = string
+            if (string.contains("twitter.com")) {
+                overrideString = string.replace("twitter.com", domain.toString())
+            } else if (string.contains("x.com")) {
+                overrideString = string.replace("x.com", domain.toString())
+            }
+
+            this.mutableMethod.replaceInstruction(
+                stringIndex,
+                """
+                    const-string v$overrideRegister, "$overrideString"
+                """
+            )
+        } ?: throw LinkBuilderMethodFingerprint.exception
+
+        LinkResourceGetterFingerprint.result?.apply {
+            val instructions = this.mutableMethod.getInstructions()
+
+            // Result register of the original method call
+            var resultRegister = 0
+
+            // Remove instructions at the end and get the register of the last instruction
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (methodRef.definingClass != "Landroid/content/res/Resources;") continue
+
+                    // Get the result register of the original method call
+                    resultRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+
+                    // Remove the original method call for getting the link from resources and the move-result-object instruction
+                    this.mutableMethod.removeInstructions(index, 2)
+
+                    // Remove the instruction that uses the resultRegister as an array reference to prevent an error
+                    this.mutableMethod.removeInstructions(index - 2, 1)
+
+                    break
+                }
+            }
+
+            // Save user nickname to v7
+            for ((index, instruction) in instructions.withIndex()) {
+                if (instruction.opcode == Opcode.INVOKE_VIRTUAL) {
+
+                    val methodRef =
+                        (instruction as ReferenceInstruction).reference as MethodReference
+                    if (
+                        methodRef.returnType != "Ljava/lang/String;") continue
+
+                    val sourceRegister = (instructions[index + 1] as OneRegisterInstruction).registerA
+                    this.mutableMethod.addInstruction(
+                        index + 2,
+                        """
+                            move-object v7, v$sourceRegister
```

Comment:
The register could change over versions, this should not be hardcoded.

Reply:
This is a register that I use as a temporary storage, so it probably won't change

Reply:
It's just everything is used there and I'm not sure how to do that

Reply:
'v2' might be a good candidate. v2 is assigned to 'this'. And later used. But I can get this register, overwrite it with my string and then restore it after using it by copying the instruction. Relevant change is added

Reply:
Hopefully nothing is gonna use 'this' between two lines of code